internal/pkg/service/stream/storage/statistics/repository/stats_get_test.go

```diff
@@ -0,0 +1,363 @@
+		require.NoError(t, defRepo.Branch().Create(&branch, clk.Now(), by).Do(ctx).Err())
+		source := test.NewSource(sourceKey)
+		require.NoError(t, defRepo.Source().Create(&source, clk.Now(), by, "Create source").Do(ctx).Err())
+		sink := test.NewSinkWithLocalStorage(sinkKey)
+		require.NoError(t, defRepo.Sink().Create(&sink, clk.Now(), by, "Create sink").Do(ctx).Err())
+	}
+
+	// Create 2 more files/slices (3 totally)
+	// -----------------------------------------------------------------------------------------------------------------
+	var sliceKey1, sliceKey2, sliceKey3 model.SliceKey
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		slices, err := sliceRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, slices, 3)
+		sliceKey1 = slices[0].SliceKey
+		sliceKey2 = slices[1].SliceKey
+		sliceKey3 = slices[2].SliceKey
+	}
+
+	// Empty
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		v, err := statsRepo.ProjectStats(ctx, sliceKey1.ProjectID)
+		assert.Empty(t, v)
+		assert.NoError(t, err)
+		v, err = statsRepo.SourceStats(ctx, sliceKey1.SourceKey)
+		assert.Empty(t, v)
+		assert.NoError(t, err)
+		v, err = statsRepo.SinkStats(ctx, sliceKey1.SinkKey)
+		assert.Empty(t, v)
+		assert.NoError(t, err)
+		v, err = statsRepo.FileStats(ctx, sliceKey1.FileKey)
+		assert.Empty(t, v)
+		assert.NoError(t, err)
+		v, err = statsRepo.SliceStats(ctx, sliceKey1)
+		assert.Empty(t, v)
+		assert.NoError(t, err)
+	}
+
+	// Add some statistics
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		assert.NoError(t, statsRepo.Put(ctx, []statistics.PerSlice{
+			{
+				SliceKey: sliceKey1,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T01:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T02:00:00.000Z"),
+					RecordsCount:     1,
+					UncompressedSize: 1,
+					CompressedSize:   1,
+				},
+			},
+			{
+				SliceKey: sliceKey2,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T02:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T03:00:00.000Z"),
+					RecordsCount:     10,
+					UncompressedSize: 10,
+					CompressedSize:   10,
+				},
+			},
+			{
+				SliceKey: sliceKey3,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T03:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T04:00:00.000Z"),
+					RecordsCount:     100,
+					UncompressedSize: 100,
+					CompressedSize:   100,
+				},
+			},
+		}))
+	}
+
+	// Move statistics file2 to the staging, and file3 to the target level
```

Comment:
How it works. You move file2 to staging and file3 to target level but actualy you switched the slice. Is the commentary correct? I feel like there is a lot of copy paste issues according comments part that do not align the code.

Reply:
I will check it,

> I feel like there is a lot of copy paste issues according comments

Yes, because there is a lot of comments :)

Reply:
Finally, I see no typo in comments yet, I am adding one more comment there:

```
// All slices must be marked uploaded, to mark the file as imported.
```

Reply:
[fix: Review fixes](https://github.com/keboola/keboola-as-code/pull/1820/commits/03b95f2f3ba09bd1292215856802fa207b2b71aa)

Reply:
TODO: link mapping between state -> storage level

Reply:
See [FileState.ToLevel](https://github.com/keboola/keboola-as-code/blob/f9413c4782852cef5909dce70146b34c7dc6156e/internal/pkg/service/stream/storage/model/filestate.go#L74-L86) and [SliceState.ToLevel](https://github.com/keboola/keboola-as-code/blob/f9413c4782852cef5909dce70146b34c7dc6156e/internal/pkg/service/stream/storage/model/slicestate.go#L62-L74).